### PR TITLE
Gui: Fix PythonWrapper::toEnum() to also accept a Python int as argument

### PR DESCRIPTION
**Pull request by wwmayer**
See https://github.com/FreeCAD/FreeCAD/pull/13611#issuecomment-2097126565

### CODE REVIEW

**Comment by wwmayer** [comment]
This PR fixes toEnum() to support `getStandardButtons()` returning an int or a `QtGui.QDialogButtonBox.StandardButton`.

The task dialog can be:
```
from PySide import QtGui

class TaskPanel:
    def getStandardButtons(self):
        return int(QtGui.QDialogButtonBox.Ok | QtGui.QDialogButtonBox.Cancel)

panel = TaskPanel()
FreeCADGui.Control.showDialog(panel)
```
or
```
from PySide import QtGui

class TaskPanel:
    def getStandardButtons(self):
        return QtGui.QDialogButtonBox.Ok | QtGui.QDialogButtonBox.Cancel

panel = TaskPanel()
FreeCADGui.Control.showDialog(panel)
```

File: src/Gui/PythonWrapper.cpp [hunk 1]
```diff
@@ -598,17 +601,19 @@ qsizetype PythonWrapper::toEnum(const Py::Object& pyobject)
...
+        qsizetype ret {};
         if (pyobject.hasAttr(std::string("value"))) {
             ret = Py::Int(pyobject.getAttr(std::string("value")));
-        } else {
+        }
+        else {
             ret = Py::Int(pyobject);
         }
-        return (qsizetype)ret;
+        return ret;
```
Comment (3x380V) [inline src/Gui/PythonWrapper.cpp:611]: This seems to be enough:
```
        if (pyobject.hasAttr(std::string("value"))) {
            return Py::Int(pyobject.getAttr(std::string("value")));
        }
        return Py::Int(pyobject);
```

**Review by YakoYakoYokuYoku — APPROVED** [review]
Besides other feedback from reviewers this LGTM and thanks for the fix. By the way, I think that the close paren is still missing but that should be easy to solve.

**Comment by wwmayer** [comment]
> I think that the close paren is still missing but that should be easy to solve

No, this fix is already merged into main.

**Pull request merged**